spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -303,13 +304,13 @@ public RetryTemplate buildRetryTemplate(ConsumerProperties properties) {
 		private static ConcurrentMap<String, MimeType> mimeTypesCache = new ConcurrentHashMap<>();
 
 		static MimeType mimeTypeFromObject(Object payload, String originalContentType) {
+
 			Assert.notNull(payload, "payload object cannot be null.");
 			if (payload instanceof byte[]) {
 				return MimeTypeUtils.APPLICATION_OCTET_STREAM;
 			}
 			if (payload instanceof String) {
-				return MimeTypeUtils.APPLICATION_JSON_VALUE.equals(originalContentType) ? MimeTypeUtils.APPLICATION_JSON
-						: MimeTypeUtils.TEXT_PLAIN;
+				return originalContentType.toLowerCase().contains(MimeTypeUtils.APPLICATION_JSON.getSubtype()) ? MimeType.valueOf(originalContentType) : MimeTypeUtils.TEXT_PLAIN;
```

Comment:
You know have opened the pandora box my friend :)

My humble opinion is that we are mixing a little bit of content negotiation and serialization on our current model. And perhaps it's what causes a bit of confusion and disconnection on our discussion

I think we will benefit from a f2f call, I'll send an invite to all members of the team that are interested in chiming in.

I have an opinion that we should only really send byte[], not even care if payload is of type String. If contentType proves to be a String representation (text/plain, application/csv) we could convert to String.class, otherwise most converters such as jackson are ok in reading byte[].

I am pretty sure my view is limited and will be proven wrong, but I'm glad to get educated on this.